src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -129,6 +129,42 @@ public async Task<bool> RestoreAsync(string entryProjectFilePath, IDictionary<st
             }
         }
 
+        /// <summary>
+        /// Generates a dependency graph spec for the given properties.
+        /// </summary>
+        /// <param name="entryProjectFilePath">The main project to generate that graph for.  This can be a project for a Visual Studio© Solution File.</param>
+        /// <param name="globalProperties">The global properties to use when evaluation MSBuild projects.</param>
+        /// <param name="options">The set of options to use to generate the graph, including the restore graph output path.</param>
+        /// <returns><code>true</code> if the dependency graph spec was generated and written, otherwise <code>false</code>.</returns>
+        public bool WriteDependencyGraphSpec(string entryProjectFilePath, IDictionary<string, string> globalProperties, IReadOnlyDictionary<string, string> options)
+        {
+            var dependencyGraphSpec = GetDependencyGraphSpec(entryProjectFilePath, globalProperties);
+
+            try
+            {
+                if (dependencyGraphSpec == null)
+                {
+                    LoggingQueue.TaskLoggingHelper.LogError(Strings.Error_DgSpecGenerationFailed);
+                    return false;
+                }
+
+                if (options.TryGetValue("RestoreGraphOutputPath", out var path))
+                {
+                    dependencyGraphSpec.Save(path);
+                    return true;
+                }
+                else
+                {
+                    LoggingQueue.TaskLoggingHelper.LogError(Strings.Error_MissingRestoreGraphOutputPath);
+                }
```

Comment:
Check `RestoreGraphOutputPath` before calling `GetDependencyGraphSpec`, so this error case doesn't need to do a potentially expensive operation.

Reply:
The target itself will check that, so there's a quick out already.